allennlp/training/metrics/spearman_correlation.py

```diff
@@ -40,13 +40,16 @@ def __call__(
         mask : `torch.Tensor`, optional (default = None).
             A tensor of the same shape as `predictions`.
         """
-        predictions, gold_labels, mask = self.unwrap_to_tensors(predictions, gold_labels, mask)
-        # Flatten predictions, gold_labels, and mask. We calculate the spearman correlation between
+        predictions, gold_labels, mask = self.detach_tensors(predictions, gold_labels, mask)
+        # Flatten predictions, gold_labels, and mask. We calculate the Spearman correlation between
         # the vectors, since each element in the predictions and gold_labels tensor is assumed
         # to be a separate observation.
         predictions = predictions.view(-1)
         gold_labels = gold_labels.view(-1)
 
+        self.total_predictions = self.total_predictions.to(predictions.device)

```

Comment:
Note that at initialization time we don't know the device we should use, but here we move it. If it was already in that device, it's a no-op so it's fine.